syz-ci/manager.go

```diff
@@ -910,14 +911,22 @@ func (mgr *Manager) uploadCoverJSONLToGCS(mgrSrc, gcsDest string, curTime time.T
 	pr, pw := io.Pipe()
 	defer pr.Close()
 	go func() {
+		var closeError error
+		defer func() { pw.CloseWithError(closeError) }()
+		var w io.Writer
+		w = pw
+		if compress {
+			gzw := gzip.NewWriter(pw)
+			defer gzw.Close()
+			w = gzw
+		}
 		decoder := json.NewDecoder(resp.Body)
 		for decoder.More() {
-			if err := f(pw, decoder); err != nil {
-				pw.CloseWithError(fmt.Errorf("callback: %w", err))
+			if err := f(w, decoder); err != nil {
+				closeError = fmt.Errorf("callback: %w", err)
```

Comment:
It is a bit implicit that this assignment is used in the deferred function.
Let's hear what @a-nogikh says

Reply:
In the last commit here, I'm adding the context to uploadFile.
We can stop upload using this context. I can add one more optimizing commit on top of what you see here.

Reply:
@ramosian-glider ptal last commit.